### PR TITLE
REF: support reso in remaining tslibs helpers

### PR DESCRIPTION
**Pull request by jbrockmendel**
Mostly just to trim the diff in upcoming PRs.

### CODE REVIEW

File: pandas/_libs/tslib.pyx [hunk 1]
```diff
@@ -95,7 +95,7 @@ def _test_parse_iso8601(ts: str):
         return Timestamp.now().normalize()
 
     string_to_dts(ts, &obj.dts, &out_bestunit, &out_local, &out_tzoffset, True)
-    obj.value = dtstruct_to_dt64(&obj.dts)
+    obj.value = npy_datetimestruct_to_datetime(NPY_FR_ns, &obj.dts)
```
Comment (mroeschke) [inline pandas/_libs/tslib.pyx:98]: FWIW: It looks like this entire function is just unit tested and not used anywhere else so could be removed?

File: pandas/_libs/tslibs/np_datetime.pxd [hunk 2]
```diff
@@ -75,11 +75,13 @@ cdef bint cmp_scalar(int64_t lhs, int64_t rhs, int op) except -1
 
 cdef check_dts_bounds(npy_datetimestruct *dts, NPY_DATETIMEUNIT unit=?)
 
-cdef int64_t dtstruct_to_dt64(npy_datetimestruct* dts) nogil
-
-cdef int64_t pydatetime_to_dt64(datetime val, npy_datetimestruct *dts)
+cdef int64_t pydatetime_to_dt64(
+    datetime val, npy_datetimestruct *dts, NPY_DATETIMEUNIT reso=?
```
Comment (mroeschke) [inline pandas/_libs/tslibs/np_datetime.pxd:79]: No defaulting `reso` to `NPY_DATETIMEUNIT.NPY_FR_ns`?

File: pandas/_libs/tslib.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslib.pyx:98]: yah, i always figured it was really a test for string_to_dts, would be OK with declaring we already have enough indirect tests of that

File: pandas/_libs/tslibs/np_datetime.pxd (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/np_datetime.pxd:79]: we'll soon need to be explicit everywhere this is called anyway

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**